HotelApi/Controllers/ReservationController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Web.Http;
+using System.Web.Http.Results;
+using HotelBooking;
+using HotelBooking.BookingExternalService;
+using HotelBooking.HotelFactory;
+using HotelBooking.HotelManager;
+using HotelBooking.Logger;
+using HotelBooking.PaymentExternalService;
+
+namespace HotelApi.Controllers
+{
+    public class ReservationController : ApiController
+    {
+        private readonly HotelManager _manager;
+
+        public ReservationController()
+        {
+            IHotelFactory factory = new HotelFactory();
+            IPaymentService paymentService = new PaymentService();
+            IBookingService bookingService = new BookingService();
+            ILogger logger = new ConsoleLogger();
+            _manager = new HotelManager(factory, paymentService, bookingService, logger);
+        }
+
+        [HttpGet]
+        public JsonResult<List<int>> GetAvailableHotelsIds()
+        {
+            return Json(_manager.PresentAvailableHotels());
+        }
+
+        [HttpPost]
+        public JsonResult<ReservationResult> MakeReservation(int hotelId, double price, int creditCardNumber,
```

Comment:
Should consider:

If it is a json api or one of the supported format is json why not use it to deserialization as well. It is mean that instead of multiple attributes you can create object e.g. ReservationRequest, which  may be simple DTO with DataContract attributes to handle serialization properly. It is a lot of stuff in google [just one of the first](https://stackoverflow.com/questions/22492821/webapi-deserializing-and-serializing-alternate-property-names)

Reply:
applied